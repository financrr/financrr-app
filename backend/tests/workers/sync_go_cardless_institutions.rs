use financrr::{
    app::App,
    workers::sync_go_cardless_institutions::{SyncGoCardlessInstitutionsWorker, WorkerArgs},
};
use loco_rs::{bgworker::BackgroundWorker, testing::prelude::*};
use serial_test::serial;

#[tokio::test]
#[serial]
async fn test_run_sync_go_cardless_institutions_worker() {
    let boot = boot_test::<App>().await.unwrap();

    // Execute the worker ensuring that it operates in 'ForegroundBlocking' mode, which prevents the addition of your worker to the background
    assert!(
        SyncGoCardlessInstitutionsWorker::perform_later(&boot.app_context, WorkerArgs {})
            .await
            .is_ok()
    );
    // Include additional assert validations after the execution of the worker
}
