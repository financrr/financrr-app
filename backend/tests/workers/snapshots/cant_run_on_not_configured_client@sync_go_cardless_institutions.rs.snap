---
source: tests/workers/sync_go_cardless_institutions.rs
assertion_line: 33
expression: rs.err().unwrap()
---
Any(
    AppError {
        status_code: 500,
        error_code: ErrorCode {
            code: 1013,
            message: "An error in the server configuration.",
        },
        details: None,
        reference: Some(
            JsonReference {
                payload: String("GoCardless client not configured."),
            },
        ),
    },
)
